server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -168,16 +176,24 @@ public int getNumOpenConnections()
       final QueryMetrics<? super Query<T>> queryMetrics = toolChest.makeMetrics(query);
       queryMetrics.server(host);
 
+      long timeoutAt = ((Long) context.get(QUERY_FAIL_TIME)).longValue();
```

Comment:
Druid SQL ran into that too and was fixed via #4305.

I think this is fine since, imo, making queries without going through the resources is something you do 'at your own risk' and isn't an officially supported mode of operation.